.buildkite/scripts/common.sh

```diff
@@ -95,6 +95,94 @@ with_Terraform() {
     terraform version
 }
 
+fix_gsutil() {
+    # this function checks if gsutil is working and if not, replaces it with a Python virtualenv version (and installs a supported Python version without affecting the system Python)
+    # TODO remove once the core ubuntu 20.04 aarch64 images gets fixed to include a working gsutil
+
+    if gsutil --version >/dev/null 2>&1; then
+        echo "--- gsutil works; nothing to do."
+    else
+        echo "--- gsutil doesn't work -- uninstalling it and installing a supported Python version and gsutil via virtualenv..."
+        # Remove apt-based Cloud SDKs if present
+        sudo apt-get update || true
+        sudo DEBIAN_FRONTEND=noninteractive apt-get -y purge google-cloud-cli google-cloud-sdk || true
+
+        # Remove common archive-install locations and stray shims
+        sudo rm -rf /opt/google-cloud-sdk /usr/lib/google-cloud-sdk || true
+        sudo rm -f /usr/bin/gsutil /usr/local/bin/gsutil /usr/bin/gcloud /usr/local/bin/gcloud || true
+
+        SDK_DIR="/opt/google-cloud-sdk"
+        VENV="/opt/gcloud-py"
+        CONFIG_DIR="/opt/buildkite-agent/.config/gcloud"
+        
+        # Set architecture-specific archive URL
+        check_platform_architeture
+        case "${arch_type}" in
+            "amd64")
+                ARCHIVE_URL="https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-cli-linux-x86_64.tar.gz"
+                ;;
+            "arm64")
+                ARCHIVE_URL="https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-cli-linux-arm.tar.gz"
+                ;;
+            *)
+                echo "Unsupported architecture: ${arch_type}"
+                return 1
+                ;;
+        esac
+
+        # Ensure config dir exists and is writable by the current user
+        sudo mkdir -p "$CONFIG_DIR"
+        sudo chown -R "$(id -u)":"$(id -g)" "$CONFIG_DIR"
+
+        # Install a modern Python alongside (3.10) -- this doesn't affect the system Python
+        sudo apt-get update -y
```

Comment:
nit: I would also export `DEBIAN_FRONTEND=noninteractive` just to be safe. In theory `-y` does the same but I have faint recollections from past experience that there are some cases that the env var is still needed. Feel free to ignore